lib/user_data_parser/row/row.rb

```diff
@@ -0,0 +1,31 @@
+require_relative "../utils/date.rb"
+
+module UserDataParser
+  class Row
+    def initialize(data)
+      set_columns(data)
+    end
+
+    def set_columns(data)
+      data.each do |k,v|
+        singleton_class.class_eval { attr_reader "#{k}" }
```

Comment:
Nitpick: consider encapsulating this in a private method so it's more clear what's going on here. What are we actually setting on the columns here?